### PR TITLE
Add process to change membership category to partner window

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_Change_Product_for_Partner.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	final ProductCategoryId retrieveProductCategoryId()
+	{
+		return productDAO.retrieveProductCategoryByProductId(retrieveSelectedProductId());
+	}
+
+	final BPartnerId retrieveSelectedBPartnerId()
+	{
+		return BPartnerId.ofRepoId(getRecord_ID());
+	}
```
Comment (adi-stefan) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_Change_Product_for_Partner.java:124]: these shall all be private methods

**Review by adi-stefan — CHANGES_REQUESTED** [review]
please see my comments

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java [hunk 2]
```diff
@@ -2076,7 +2076,7 @@ private void updateProductForInvoiceCandidate(@NonNull final FlatrateTermPriceRe
 
 		final I_C_Invoice_Candidate ic = flatrateDAO.retrieveInvoiceCandidate(term);
 
-		if (ic == null)
+		if (ic == null && ic.isProcessed())
```
[the remark this answers is not in the document]
Comment (cristinamghita) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java:2079]: I wanted ||

File: backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5705130_sys_AddNewProcessForUpdatingProductsInContracts.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+UPDATE AD_Process_Para SET EntityType='D',Updated=TO_TIMESTAMP('2023-10-04 18:09:40','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Process_Para_ID=542706
+;
+
+-- 2023-10-04T15:10:12.393Z
+INSERT INTO AD_Table_Process (AD_Client_ID,AD_Org_ID,AD_Process_ID,AD_Table_ID,AD_Table_Process_ID,Created,CreatedBy,EntityType,IsActive,Updated,UpdatedBy,WEBUI_DocumentAction,WEBUI_IncludedTabTopAction,WEBUI_ViewAction,WEBUI_ViewQuickAction,WEBUI_ViewQuickAction_Default) VALUES (0,0,585322,291,541429,TO_TIMESTAMP('2023-10-04 18:10:12','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',TO_TIMESTAMP('2023-10-04 18:10:12','YYYY-MM-DD HH24:MI:SS'),100,'Y','N','Y','Y','N')
+;
+
+-- 2023-10-05T12:37:53.788Z
+UPDATE AD_Val_Rule SET Code='M_Product.AD_Org_ID = (select AD_Org_ID from C_BPartner bp where bp.C_BPartner_id = @C_BPartner_ID@) ',Updated=TO_TIMESTAMP('2023-10-05 15:37:53','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Val_Rule_ID=540659
+;
```
[the remark this answers is not in the document]
Comment (cristinamghita) [reply backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5705130_sys_AddNewProcessForUpdatingProductsInContracts.sql:36]: yes

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_Change_Product_for_Partner.java (hunk 1 above)
Comment (cristinamghita) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/flatrate/process/C_Flatrate_Term_Change_Product_for_Partner.java:124]: there are. not explicitly set

**Review by adi-stefan — APPROVED** [review]

**Pull request merged**